.github/workflows/python-app-macos.yml

```diff
@@ -0,0 +1,80 @@
+# This workflow will install Python dependencies, run tests
+
+name: Python application
+
+on:
+  push:
+    branches: [ "main" , "dev"]
+    paths-ignore:
+      - "docs/**"
+      - "**/README.md"
+
+  pull_request:
+    branches: [ "main" , "dev"]
+    paths-ignore:
+      - "docs/**"
+      - "**/README.md"
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  build:
+    env:
+      DISPLAY: :0
+  
+    defaults:
+      run:
+        shell: bash
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - macos-latest
+
+    steps:
+    - name: Checkout repository and submodules
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"     
+
+    - name: Enable Conda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        activate-environment: anaconda-client-env
+        auto-activate-base: false
+    - name: Test Conda
+      run: |
+          conda info
+          CONDA_DEFAULT_ENV=test_myosuite
+          conda create --name $CONDA_DEFAULT_ENV python=3.8 -y
+          source /usr/local/miniconda/etc/profile.d/conda.sh
```

Comment:
wouldn't this work in Win/Linux?

Reply:
nope, in Windows doesn't work while in Linux the path is different. Of course the single-file is doable but the resulting workflow would be verbose, at least with this solution.